crates/core/c8y_api/src/smartrest/operations.rs

```diff
@@ -127,6 +140,29 @@ impl Operations {
             .collect::<HashSet<String>>()
             .try_into()?)
     }
+
+    /// Return operation and template name if `command_name` matches
+    pub fn match_template_with_operation_by_command_name(&self, command: &str) -> Option<String> {
+        self.templates
+            .iter()
+            .filter(|template| {
+                template
+                    .command_name()
+                    .is_some_and(|command_name| command_name.eq(command))
+            })
+            .find_map(|template| template.on_fragment())
```

Comment:
This makes me wonder if the `on_fragment` filed is redundant, when the name of the operation file should have been be the same. Are there any cases in C8y where the operation name and this fragment name could be different?

Reply:
I wouldn't that the fragment and the name of the operation are the same. These are custom operations which means that the customer is free to configure them however they want, so being more explicit would be safer here with the downside of a bit of boilerplate.

Reply:
@Ruadhri17 if you're also wondering why any user would want to declare one name in the supported operations list (the operation file name) and use an entirely different name for his operation fragment, here is the use-case: The declared supported operation name is typically used to enable/disable UI components in C8y. The developer of that UI component dictates the expected operation name. And it is a possibility that the custom operation developer may not be able to (or don't want to) reuse the same name for his operation as well. In such cases, this design would help.